assembly/src/parsers/io_ops/mem_ops.rs

```diff
@@ -47,6 +56,10 @@ pub fn parse_pop_mem(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), As
 /// Then, if the memory address was provided via the stack (not as part of the memory op) it must be
 /// moved to the top.
 ///
+/// This operation takes:
+///  - pushw: 6 VM cycles.
+///  - loadw: 2 / 1 VM cycles (immediate value / via the stack).
```

Comment:
Yes, in case of an immediate value we need to push it on the top of the stack, but if it was already on the stack we need to move it up after `Pad` operations. So we need to perform one operation anyway.

Reply:
Got it! Thank you!